### ISSUE TITLE
[csharp] Codegen for enum models results in broken code

### ISSUE THREAD

**Issue by Inirit** [swagger-api/swagger-codegen#7656]
##### Description

When generating models for a C# client I've found that the resulting code for enums is broken (i.e. does not compile). I've also found that it only repros given certain conditions, but these conditions are met frequently and easily enough that this issue would prevent my use of the codegen tool entirely. 

##### Swagger-codegen version

I discovered the issue using version 2.3.1. Performing the same actions with the same inputs **did not repro using version 2.3.0**, so it would appear to be a recent regression.

##### Swagger declaration file content or url

https://gist.github.com/Inirit/1e5c61ecdfb1f5f9b1b3cd33ae86f2ab

##### Command line used for generation

This command will repro the issue with minimal noise
```
java -Dmodels -DmodelDocs=false -jar swagger-codegen-cli-2.3.1.jar generate -l csharp -i myspec.json -o output_2.3.1
```

This similar command will **only generate the enum model**, but **notice that this time the resulting code is valid** (though with some odd excess blank lines). It seems to imply that the enum codegen will only fail if the enum is referenced by another model in the same spec.
```
java -Dmodels=MyEnum -DmodelDocs=false -jar swagger-codegen-cli-2.3.1.jar generate -l csharp -i myspec.json -o output_2.3.1_enum_only
```

This command is also similar to the first, except it uses version 2.3.0 of the codegen tool. Unlike the first command, this time **the enum code is valid**. And unlike the second command, this time **it was able to handle the entire spec** (and also didn't include excess blank lines)
```
java -Dmodels -DmodelDocs=false -jar swagger-codegen-cli-2.3.0.jar generate -l csharp -i myspec.json -o output_2.3.0
```

##### Steps to reproduce

1. Use the spec provided by the above Gist
2. Use the first command provided above
3. After codegen is complete, navigate down to the generated MyEnum.cs and open it in a text editor
4. Observe the status of the file, the code is invalid and will not compile

##### Related issues/PRs

I looked around but didn't see anything specifically about this.

##### Suggest a fix/enhancement

Restore functionality, matching 2.3.0

**Comment by ttalle** [comment]
I don't have a development environment at hand but with a quick browse through the code I found a line that could be the source of the problem:

Shouldn't the line:
https://github.com/swagger-api/swagger-codegen/blob/3f3d9c7f2a906fdafa27ff2b7b33a552f654f9b2/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractCSharpCodegen.java#L926

be something like:

`return "\"" + escapeText(value) + "\"";`

**Comment by ffes** [comment]
Run into this problem as well.

Could some maintainer please add the c-sharp label to this issue?

**Comment by frantuma** [comment]
should be fixed in #9936
